tests/test_models_circuit.py

```diff
@@ -375,6 +378,32 @@ def test_circuit_serialization():
     assert Circuit.from_dict(raw).raw == raw
 
 
+def test_circuit_serialization_with_wire_names():
+
+    wire_names = ["a", "b"]
+    c = Circuit(2, wire_names=wire_names)
+    raw = c.raw
+    assert "wire_names" in raw
+    new_c = Circuit.from_dict(raw)
+    assert new_c.wire_names == c.wire_names
+
+    get_backend()  # this has to be called before get_transpiler
+    # otherwise it crashes
+    transpiler = get_transpiler()
+    g = Graph()
+    g.add_edge(*wire_names)
+    transpiler.connectivity = g
+    transpiler.passes = [Sabre()]
+
+    c, _ = transpiler(c)
+    new_c, _ = transpiler(new_c)
+    assert new_c.wire_names == c.wire_names
```

Comment:
> In the end I just directly triggered the transpiler error as globally setting a mock backend is not straightforward.

That is perfectly fine, but at this point you do not need to pass through a backend at all
```suggestion
    transpiler = Passes(passes=[Sabre()], connectivity=Graph([wire_names]))

    c, _ = transpiler(c)
    new_c, _ = transpiler(new_c)
    assert new_c.wire_names == c.wire_names
```

Reply:
ah ok, so without getting the transpiler with ``get_transpiler()``, it makes sense.